py-polars/tests/unit/test_exprs.py

```diff
@@ -1034,3 +1034,135 @@ def test_is_not_deprecated() -> None:
 
     expected = pl.DataFrame({"a": [False, True, False]})
     assert_frame_equal(result, expected)
+
+
+def test_cut() -> None:
+    df = pl.DataFrame({"foo": [-2, -1, 0, 1, 2]})
+    out = df.with_columns(
+        pl.col("foo").cut([-1, 1], labels=["a", "b", "c"]).alias("cut")
+    )
+    expected = pl.DataFrame(
+        {
+            "foo": [-2, -1, 0, 1, 2],
+            "cut": pl.Series(["a", "a", "b", "b", "c"], dtype=pl.Categorical),
+        }
+    )
+    assert_frame_equal(out, expected, categorical_as_str=True)
+
+    # test with breaks
+    expected = pl.DataFrame(
+        {
+            "foo": [-2, -1, 0, 1, 2],
+            "brk": [-1.0, -1.0, 1.0, 1.0, float("inf")],
+            "foo_bin": pl.Series(
+                [
+                    "(-inf, -1]",
+                    "(-inf, -1]",
+                    "(-1, 1]",
+                    "(-1, 1]",
+                    "(1, inf]",
+                ],
+                dtype=pl.Categorical,
+            ),
+        }
+    )
+
+    # eager
+    out = df.with_columns(
+        pl.col("foo").cut([-1, 1], include_breaks=True).alias("cut")
+    ).unnest("cut")
+    assert_frame_equal(out, expected, categorical_as_str=True)
+
+    # lazy
+    out = (
+        df.lazy()
+        .with_columns(pl.col("foo").cut([-1, 1], include_breaks=True).alias("cut"))
```

Comment:
You should add a test that checks the schema at this point, e.g.:

```python
result.schema = {"cut": pl.Struct(...)}
```

Same for qcut.

Reply:
Doesn't `assert_frame_equal` include a schema check anyway? I'll add one regardless.

Reply:
It will check the schema, but it will collect first. The lazy schema can be different before/after collection if we did not specify the lazy schema correctly. Now that I think of it, we should probably update `assert_frame_equal` to check the schema before collecting.

Reply:
> we should probably update `assert_frame_equal` to check the schema before collecting.

Done ;)